python/semantic_kernel/connectors/openapi/sk_openapi.py

```diff
@@ -0,0 +1,301 @@
+import logging
+from typing import Dict, Mapping, Optional, Union
+from urllib.parse import urljoin
+
+import aiohttp
+import requests
+from openapi_core import Spec, unmarshal_request
+from openapi_core.contrib.requests import RequestsOpenAPIRequest
+from openapi_core.exceptions import OpenAPIError
+from prance import ResolvingParser
+
+from semantic_kernel import Kernel, SKContext
+from semantic_kernel.orchestration.sk_function_base import SKFunctionBase
+from semantic_kernel.skill_definition import sk_function, sk_function_context_parameter
+from semantic_kernel.utils.null_logger import NullLogger
+
+
+class PreparedRestApiRequest:
+    def __init__(self, method: str, url: str, params=None, headers=None, payload=None):
+        self.method = method
+        self.url = url
+        self.params = params
+        self.headers = headers
+        self.payload = payload
+
+    def __repr__(self):
+        return (
+            "PreparedRestApiRequest("
+            f"method={self.method}, "
+            f"url={self.url}, "
+            f"params={self.params}, "
+            f"headers={self.headers}, "
+            f"payload={self.payload})"
+        )
+
+    def validate_request(self, spec: Spec, logger: logging.Logger = NullLogger()):
+        request = requests.Request(
+            self.method,
+            self.url,
+            params=self.params,
+            headers=self.headers,
+            json=self.payload,
+        )
+        openapi_request = RequestsOpenAPIRequest(request=request)
+        try:
+            unmarshal_request(openapi_request, spec=spec)
+            return True
+        except OpenAPIError as e:
+            logger.debug(f"Error validating request: {e}", exc_info=True)
+            return False
+
+
+class RestApiOperation:
+    def __init__(
+        self,
+        id: str,
+        method: str,
+        server_url: str,
+        path: str,
+        summary: Optional[str] = None,
+        description: Optional[str] = None,
+        parameters: Optional[Mapping[str, str]] = None,
```

Comment:
is there a reason for using mapping over dict here?

Reply:
this says to use mapping over dict for argument type hints https://docs.python.org/3/library/typing.html#typing.Dict

at this point we know we're passing in a dict so it's not really a big deal but that's why I was defaulting to mapping for arguments.